broadcast/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+//! Replication of messages across a network.
+
+use bytes::Bytes;
+use commonware_cryptography::{Digest, PublicKey};
+use futures::channel::oneshot;
+use std::future::Future;
+
+pub mod linked;
+
+/// Proof is a blob that attests to some data.
+pub type Proof = Bytes;
+
+/// Broadcaster is the interface responsible for replication of messages across a network.
+pub trait Broadcaster {
```

Comment:
This confused me because the `Broadcaster` in consensus is flipped (`consensus` "calls" broadcaster, not the application calls `broadcaster`).